manta-js/package/sdk.ts

```diff
@@ -0,0 +1,604 @@
+        x.note.incoming_note.tag = tag;
+        x.note.incoming_note.ephemeral_public_key = pk;
+        x.note.incoming_note.ciphertext = arr1;
+        delete x.note.incoming_note.header;
+
+        const lightPk = x.note.light_incoming_note.ciphertext.ephemeral_public_key;
+        // ciphertext is [u8; 96] on manta-rs, but runtime side is [[u8; 32]; 3]
+        const lightCipher = x.note.light_incoming_note.ciphertext.ciphertext;
+        const lightCiper0 = lightCipher.slice(0, 32);
+        const lightCiper1 = lightCipher.slice(32, 64);
+        const lightCiper2 = lightCipher.slice(64, 96);
+        x.note.light_incoming_note.ephemeral_public_key = lightPk;
+        x.note.light_incoming_note.ciphertext = [lightCiper0, lightCiper1, lightCiper2];
+        delete x.note.light_incoming_note.header;
+
+        // convert asset value to [u8; 16]
+        x.utxo.public_asset.value = new BN(x.utxo.public_asset.value).toArray('le', 16);
+
+        x.full_incoming_note = x.note;
+        delete x.note;
+    }
+
+    // convert sender_posts to match runtime side
+    private convertSenderPost(x:any) {
+        const pk = x.nullifier.outgoing_note.ciphertext.ephemeral_public_key;
+        const cipher = x.nullifier.outgoing_note.ciphertext.ciphertext;
+        const ciper0 = cipher.slice(0, 32);
+        const ciper1 = cipher.slice(32, 64);
+        const outgoing = {
+            ephemeral_public_key: pk,
+            ciphertext: [ciper0, ciper1]
+        };
+        x.outgoing_note = outgoing;
+        const nullifier = x.nullifier.nullifier.commitment;
+        x.nullifier_commitment = nullifier;
+        delete x.nullifier;
+    }
+
+    /// NOTE: `post` from manta-rs sign result should match runtime side data structure type.
+    private transferPost(post:any): any {
+        let json = JSON.parse(JSON.stringify(post));
+
+        // transfer authorization_signature format
+        if(json.authorization_signature != null){
+            const scala = json.authorization_signature.signature.scalar;
+            const nonce = json.authorization_signature.signature.nonce_point;
+            json.authorization_signature.signature = [scala, nonce];
+        }
+
+        // transfer receiver_posts to match runtime side
+        json.receiver_posts.map((x:any) => {
+            this.convertReceiverPost(x);
+        });
+
+        // transfer sender_posts to match runtime side
+        json.sender_posts.map((x:any) => {
+            this.convertSenderPost(x);
+        });
+
+        return json
+    }
+}
+
+export class MantaUtilities {
+
+    /// Returns the metadata for an asset with a given `asset_id` for the currently
+    /// connected network.
+    static async getAssetMetadata(api: ApiPromise, assetId: BN): Promise<any> {
+        const data = await api.query.assetManager.assetIdMetadata(assetId);
+        const json = JSON.stringify(data.toHuman());
+        const jsonObj = JSON.parse(json);
+        return jsonObj;
+    }
```

Comment:
As I mention elsewhere, I was wrong about putting this in utils, can put it in `MantaPrivateWallet`